### PR TITLE
2.3 Release

### PR DESCRIPTION
**Pull request by treid314**
*Do not Merge Yet*

#### What this PR does
This is the PR for the Mimir 2.3 Release process

#### Checklist

- [ ] Documentation added
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by pstibrany** [comment]
Please make sure to do any merge of release branch to `main` using Merge commit, not by using Squashing.

**Comment by pracucci** [comment]
Could you rebase again, please? :|

File: operations/helm/charts/mimir-distributed/CHANGELOG.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ Entries should be ordered as follows:
 
 Entries should include a reference to the Pull Request that introduced the change.
 
-## main / unreleased
+## 3.1.0
```
Comment (treid314) [inline operations/helm/charts/mimir-distributed/CHANGELOG.md:14]: Fixed in https://github.com/grafana/mimir/pull/2832: result of a bad merge

**Comment by pstibrany** [comment]
Obsoleted by https://github.com/grafana/mimir/pull/2833.

> Could you rebase again, please? :|

We can't rebase release branches. We could merge `main` to release-2.3 to fix conflicts, but that pulls all changes from `main`. It's better to start a new PR (like #2833 does).

**Pull request closed without merging**